generic/README.md

```diff
@@ -0,0 +1,190 @@
+# Generic Piranha 
+Piranha scans source files to delete code related to stale feature flags leading to a cleaner, safer, more performant, and more maintainable code base.
+This generic tree-sitter based implementation for Piranha makes it easy to extend it to new languages, new feature flag APIs (and their usage).
+
+
+## Motivation 
+
+Adding Piranha support for a new language requires re-implementing the entire refactoring logic for that particular language. This is time consuming and expensive to develop and maintain such nearly similar implementations.
+This implementation overcomes this problem by extracting the language specific syntactic transformations to tree-sitter query API based rewrite rules, and applying them to the input program as chains of rules.
+
+## Usage 
+Piranha can be configured to recognize different flag APIs by specifying a `rules.toml` file (and optionally an `edges.toml`). Piranha can be then perform the refactoring based on the flag behavior, which can be specified by providing `piranha_arguments.toml` . Moreover, Piranha can be configured to operate upon a new language by specifying a `/configuration/<lang-name>/rules.toml`, `/configuration/<lang-name>/edges.toml` and `/configuration/<lang-name>/scope_generators.toml`.
+
+```
+piranha 0.1.0
+
+USAGE:
+    piranha --path-to-codebase <PATH_TO_CODEBASE> --path-to-feature-flag-rules <PATH_TO_FEATURE_FLAG_RULES> --path-to-piranha-arguments <PATH_TO_PIRANHA_ARGUMENTS>
+
+OPTIONS:
+    -c, --path-to-codebase <PATH_TO_CODEBASE>
+            Path to source code folder
+
+    -f, --path-to-feature-flag-rules <PATH_TO_FEATURE_FLAG_RULES>
+            Folder containing the required configuration files
+
+    -h, --help
+            Print help information
+
+    -p, --path-to-piranha-arguments <PATH_TO_PIRANHA_ARGUMENTS>
+            Path to the file containing arguments for Piranha
+
+    -V, --version
+            Print version information
+```
+
+## Configuring Piranha  
+### Adding a flag API and specifying how it will be updated
+
+The example below shows a usage of an feature flag API (`experiment.isTreated(STALE_FLAG)`), in an `if_statement`. 
```

Comment:
Can you add the list of steps that a user has to do before getting Piranha running? And then link to a broader discussion from each step to the actual explanation of how it is done? This running example is super helpful.